src/lib/gas-refund/gas-refund-api.ts

```diff
@@ -185,34 +190,53 @@ export class GasRefundApi {
     };
   }
 
-  async _fetchMerkleData(address: string): Promise<GasRefundClaim[]> {
-    const grpDataResult: GasRefundClaim[] = (
-      await Promise.all([
-        Database.sequelize.query<GasRefundClaim>(MERKLE_DATA_SQL_QUERY, {
-          type: Sequelize.QueryTypes.SELECT,
-          replacements: {
-            address,
-            chainId: this.network,
-          },
-        }),
-        (
-          await GasRefundParticipation.findAll({
-            raw: true,
-            where: {
-              address,
-              chainId: this.network,
-              epoch: {
-                [Sequelize.Op.gte]: EPOCH_WHEN_OPTIMISM_STAKING_ENABLED,
-              },
+  async _fetchMerkleData(address: string): Promise<DistData> {
+    const oldEpochs = Database.sequelize.query<GasRefundClaim>(
+      MERKLE_DATA_SQL_QUERY,
+      {
+        type: Sequelize.QueryTypes.SELECT,
+        replacements: {
+          address,
+          chainId: this.network,
+        },
+      },
+    );
+
+    const newEpochs = GasRefundParticipation.findAll({
+      raw: true,
+      where: {
+        address,
+        chainId: this.network,
+        epoch: {
+          [Sequelize.Op.gte]: EPOCH_WHEN_OPTIMISM_STAKING_ENABLED,
+        },
+      },
+    }).then(v => {
```

Comment:
It'll be a bit cleaner if you `await newEpochs` instead of using `.then()`. You will still be running all promises in parallel, since you start `oldEpochs` promise first. That way you can also get rid of Promise.all on L239

Reply:
> Btw, data on L367 below isn't used anywhere

I would go ahead and cleanup but I hesitate while there's https://github.com/paraswap/paraswap-volume-tracker/pull/104 in progress

I know deleting unused var would be a trivial one, but I prefer to keep the diff minimal

> It'll be a bit cleaner if you await newEpochs instead of using .then(). You will still be running all promises in parallel

It'd be sorta-half-parallel in such a way. But ok, let me get rid of `then` in another way...

Reply:
[fix](https://github.com/paraswap/paraswap-volume-tracker/pull/105/commits/242c956e4eccd72c3a71c7b3ad4b30bf0ac8a76d)